test/test_magnet.cpp

```diff
@@ -438,6 +438,37 @@ TORRENT_TEST(make_magnet_uri2)
 	TEST_CHECK(magnet.find("&ws=http%3a%2f%2ffoo.com%2fbar") != std::string::npos);
 }
 
+TORRENT_TEST(make_magnet_uri_v2)
+{
+	auto ti = ::create_torrent(nullptr, "temporary", 16 * 1024, 13
+		, true, lt::create_torrent::v2_only);
+
+	std::string magnet = make_magnet_uri(*ti);
+	std::printf("%s len: %d\n", magnet.c_str(), int(magnet.size()));
+	TEST_CHECK(magnet.find("magnet:?xt=urn:btmh:1220") != std::string::npos);
```

Comment:
Why require the v2 info-hash to be the first parameter? I don't think there is any requirement that `xt` be the first parameter for either v1 or v2.

Reply:
you're right, that's an unnecessary restriction